smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ServiceGenerator.kt

```diff
@@ -72,7 +73,7 @@ class ServiceGenerator(private val ctx: RenderingContext<ServiceShape>) {
                     SectionServiceCompanionObject,
                     context = mapOf(SectionServiceCompanionObject.ServiceSymbol to serviceSymbol)
                 ) {
-                    renderCompanionObject()
+                    renderCompanionObject(ctx.protocolGenerator != null)
```

Comment:
style

`ctx` is a property on the class, no real reason to thread through the parameter